MiscVBAFunctions/Modules/MiscDataContainers.bas

```diff
@@ -0,0 +1,61 @@
+Attribute VB_Name = "MiscDataContainers"
+Option Explicit
+
+
+Public Sub ConcatContainers(DataStructure1 As Variant, DataStructure2 As Variant)
+    If TypeOf DataStructure1 Is Collection And TypeOf DataStructure2 Is Collection Then
+        Dim I As Long
+        For I = 1 To DataStructure2.Count
+            DataStructure1.Add DataStructure2.Item(I)
+        Next
+    ElseIf TypeOf DataStructure1 Is Dictionary And TypeOf DataStructure2 Is Dictionary Then
+        Dim key As Variant
+        For Each key In DataStructure2.Keys
+            DataStructure1.Add key, DataStructure2.Item(key)
+        Next key
+    Else
+    
+        Dim errmsg As String
+        errmsg = "ConvertToTextCompare only supports type 'Dictionary and Dictionary' or 'Collection and Collection'"
+        On Error Resume Next: errmsg = errmsg & ". Got type '" & TypeName(DataStructure1) & "' and '" & TypeName(DataStructure1) & "'": On Error GoTo 0
+        Err.Raise 5, , errmsg
+        
+    End If
+End Sub
+
+
+
+Public Function JoinContainers(DataStructure1 As Variant, ByVal DataStructure2 As Variant) As Variant
```

Comment:
I would have two additional functions called `JoinCollections` and `JoinDicts` which is called by this function. So if the user knows he wants to combine dictionaries we can call `JoinDicts` directly. I think it's a bit easier also for actuaries to understand as `Container` is a new concept that needs to be understood to understand this function